### PR TITLE
Update workflows

### PR DESCRIPTION
**Pull request by sondrelg**
Introduced [poetry](https://python-poetry.org/) for packaging and dependency management.

Hope this can help simplify publishing workflows and everything to do with pypi 🚀

### CODE REVIEW

File: .github/workflows/publish_to_pypi.yml [hunk 1]
```diff
@@ -1,6 +1,8 @@
 name: Publish django-guid to PyPI and TestPyPI 📦
 
-on: push
+on:
```
Comment (sondrelg) [inline .github/workflows/publish_to_pypi.yml:3]: Not sure why this used to be triggered on push.

Comment (JonasKs) [reply .github/workflows/publish_to_pypi.yml:3]: Not sure if this release thing is new, I missed it or it wasn’t documented, but that was the only way I found to catch tags.   
I did that together with this:
```

 if: github.event_name == 'push' && startsWith(github.event.ref, 'refs/tags') && !contains(github.ref, '-rc')
```


This is way nicer though, thanks!

File: django_guid/log_filters.py [hunk 2]
```diff
@@ -14,5 +14,5 @@ def filter(self, record: LogRecord) -> bool:
         :param record: Log record
         :return: True
         """
-        record.correlation_id = GuidMiddleware.get_guid()
+        record.__setattr__('correlation_id', GuidMiddleware.get_guid())
```
Comment (JonasKs) [inline django_guid/log_filters.py:17]: I’m not sure why this change was made? What’s the benefit of using setattr here?  
If I’m missing something and it’s better I’d like it to be written like this instead:
```python
setattr(record, 'correlation_id', GuidMiddleware.get_guid())
```

File: .github/workflows/publish_to_pypi.yml [hunk 3]
```diff
@@ -1,6 +1,8 @@
 name: Publish django-guid to PyPI and TestPyPI 📦
```
Comment (JonasKs) [inline .github/workflows/publish_to_pypi.yml:1]: Seems like the possibility to push `rc` is gone now? So there's no way of publishing to test PyPi specifically?

File: .isort.cfg [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[settings]
+multi_line_output=3
+include_trailing_comma=True
+force_grid_wrap=0
+combine_as_imports=True
+use_parentheses=True
+line_length=126
```
Comment (JonasKs) [inline .isort.cfg:7]:
```suggestion
line_length=120
```
Everything else is 120 chars, and `github` diffview is max `125` chars.

File: .github/workflows/publish_to_pypi.yml (hunk 3 above)
Comment (JonasKs) [reply .github/workflows/publish_to_pypi.yml:1]: Not that I really need it I guess

Comment (sondrelg) [reply .github/workflows/publish_to_pypi.yml:1]: I've been thinking about removing test-pypi as a workflow in other projects, and just pushing the test-version locally before publishing a release. I know we generally want to replicate the "prod" job and environment when running a test, but in this case, the test releases have only ever failed for me when the package contents contain something weird, like badly formatted rst. I don't have a strong opinion about this, but what do you think of cutting the test-job altogether?

File: .isort.cfg (hunk 4 above)
Comment (sondrelg) [reply .isort.cfg]: Yeah good catch 👍

File: django_guid/log_filters.py (hunk 2 above)
Comment (sondrelg) [reply django_guid/log_filters.py:17]: Yeah agree we shouldn't use hidden methods. The original implementation is probably fine - mypy was just complaining, so I temporarily tried the suggested implementation and forgot to revert. I'll push a fix later.

File: .github/workflows/publish_to_pypi.yml (hunk 3 above)
Comment (JonasKs) [reply .github/workflows/publish_to_pypi.yml:1]: Yeah, that works for me too honestly.  
Need to update `docs/publish.rst` if it's removed though.

File: django_guid/log_filters.py (hunk 2 above)
Comment (JonasKs) [reply django_guid/log_filters.py:17]: Hmm, I'm no MyPy expert, but it's just a string, so I guess you can declare it as such?

Comment (sondrelg) [reply django_guid/log_filters.py]: Reverted to what it was :)

File: .github/workflows/publish_to_pypi.yml (hunk 3 above)
Comment (sondrelg) [reply .github/workflows/publish_to_pypi.yml]: Updated docs and removed test-section of workflow.

**Review by JonasKs — APPROVED** [review]
Awesome! Thank youu
